test/cmp_ctx_test.c

```diff
@@ -0,0 +1,870 @@
+            || !ossl_cmp_ctx_set1_caPubs(ctx, certs)
+            || !ossl_cmp_ctx_set1_extraCertsIn(ctx, certs)
+            || !ossl_cmp_ctx_set0_validatedSrvCert(ctx, X509_new())
+            || !TEST_ptr(bytes = ASN1_OCTET_STRING_new())
+            || !OSSL_CMP_CTX_set1_transactionID(ctx, bytes)
+            || !OSSL_CMP_CTX_set1_senderNonce(ctx, bytes)
+            || !ossl_cmp_ctx_set1_recipNonce(ctx, bytes)
+        )
+        goto err;
+
+    if (!TEST_true(OSSL_CMP_CTX_reinit(ctx)))
+        goto err;
+
+    /* check whether values have been reset to default in all relevant fields */
+    if (!TEST_true(ctx->status == -1
+                       && ctx->failInfoCode == -1
+                       && ctx->statusString == NULL
+                       && ctx->newCert == NULL
+                       && ctx->caPubs == NULL
+                       && ctx->extraCertsIn == NULL
+                       && ctx->validatedSrvCert == NULL
+                       && ctx->transactionID == NULL
+                       && ctx->senderNonce == NULL
+                   && ctx->recipNonce == NULL))
+        goto err;
+
+    /* this does not check that all remaining fields are untouched */
+    res = 1;
+
+ err:
+    sk_X509_pop_X509_free(certs);
+    ASN1_OCTET_STRING_free(bytes);
+    return res;
+}
+
+static int test_CTX_reinit(void)
+{
+    SETUP_TEST_FIXTURE(OSSL_CMP_CTX_TEST_FIXTURE, set_up);
+    EXECUTE_TEST(execute_CTX_reinit_test, tear_down);
+    return result;
+}
+
+static int msg_total_size = 0;
+static int msg_total_size_log_cb(const char *func, const char *file, int line,
+                                 OSSL_CMP_severity level, const char *msg)
+{
+    msg_total_size += strlen(msg);
+    return 1;
+}
+
+#define STR64 "This is a 64 bytes looooooooooooooooooooooooooooooooong string.\n"
+/* max string length ISO C90 compilers are required to support is 509. */
+#define STR509 STR64 STR64 STR64 STR64 STR64 STR64 STR64 \
+    "This is a 61 bytes loooooooooooooooooooooooooooooong string.\n"
+static const char *const max_str_literal = STR509;
+#define STR_SEP "<SEP>"
+
+static int execute_CTX_print_errors_test(OSSL_CMP_CTX_TEST_FIXTURE *fixture)
+{
+    OSSL_CMP_CTX *ctx = fixture->ctx;
+    int trace_enabled = 1;
+    int base_err_msg_size, expected_size;
+    int res = 1;
+
+#ifdef OPENSSL_NO_TRACE
+    trace_enabled = 0;
+#endif
+    if (!TEST_int_eq(OSSL_CMP_CTX_set_log_cb(ctx, NULL), trace_enabled))
+        res = 0;
+    if (!TEST_true(ctx->log_cb == NULL))
+        res = 0;
+
+#ifndef OPENSSL_NO_STDIO
+    CMPerr(0, CMP_R_MULTIPLE_SAN_SOURCES);
+    OSSL_CMP_CTX_print_errors(ctx); /* should print above error to STDERR */
+#endif
+
+    /* this should work regardless of OPENSSL_NO_STDIO and OPENSSL_NO_TRACE: */
+#ifndef OPENSSL_NO_TRACE
+    if (!TEST_true(OSSL_CMP_CTX_set_log_cb(ctx, msg_total_size_log_cb)))
+        res = 0;
+#else
+    ctx->log_cb = msg_total_size_log_cb;
+#endif
+    if (!TEST_true(ctx->log_cb != NULL))
```

Comment:
Style nit: The else uses {}, so the if block should do to.

Reply:
Fixed.